lib/pure/strutils.nim

```diff
@@ -1551,6 +1551,26 @@ proc replaceWord*(s, sub: string, by = ""): string {.noSideEffect,
   # copy the rest:
   add result, substr(s, i)
 
+proc multiReplace*(s: string, replacements: varargs[(string, string)]): string {.noSideEffect.} =
+  ## Same as replace, but optimized for doing multiple replacements in a single
+  ## pass.
+  ##
+  ## The order of the replacements does matter. Earlier replacements are preferred over later
+  ## replacements in the argument list.
+  result = newStringOfCap(s.len)
+  var charsRead = 0
+  while charsRead < s.len:
+    var noReplacement = true
```

Comment:
You can use a 'block label' + 'break label' here instead of the bool 'noReplacement' flag.